packages/common/src/types/field.ts

```diff
@@ -123,6 +123,14 @@ export type FieldId = string;
 export const fieldId = (field: Pick<Field, 'table' | 'name'>): FieldId =>
     `${field.table}_${field.name}`;
 
+export const convertFieldIdToFieldRef = (field: Field) =>
+    `${field.table}.${field.name}`;
+
+export const convertFieldRefToFieldId = (fieldRef: string) => {
+    const [table, name] = fieldRef.split('.');
+    return `${table}_${name}`;
+};
```

Comment:
We should handle the case where the reference doesn't have a table. In those cases we assume it's the same table from the metric. 
Copy the logic from `resolveQueryFieldReference` to here and delete that function.